standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/AbstractJDBCConnectorProvider.java

```diff
@@ -0,0 +1,311 @@
+    } catch (SQLException sqle) {
+      LOG.warn("Could not retrieve table names from remote datasource, cause:" + sqle.getMessage());
+    } finally {
+      try {
+        if (rs != null) {
+          rs.close();
+          rs = null;
+        }
+      } catch(Exception e) { /* ignore */}
+    }
+    return null;
+  }
+
+  protected abstract ResultSet fetchTableMetadata(String tableName) throws MetaException;
+
+  protected abstract ResultSet fetchTableNames() throws MetaException;
+
+  /**
+   * Fetch a single table with the given name, returns a Hive Table object from the remote database
+   * @return Table A Table object for the matching table, null otherwise.
+   * @throws MetaException To indicate any failures with executing this API
+   * @param tableName
+   */
+  @Override public Table getTable(String tableName) throws MetaException {
+    ResultSet rs = null;
+    Table table = null;
+    try {
+      // rs = fetchTableMetadata(tableName);
+      rs = fetchTableViaDBMetaData(tableName);
+      List<FieldSchema> cols = new ArrayList<>();
+      while (rs.next()) {
+        FieldSchema fs = new FieldSchema();
+        fs.setName(rs.getString("COLUMN_NAME"));
+        fs.setType(getDataType(rs.getString("TYPE_NAME"), rs.getInt("COLUMN_SIZE")));
+        fs.setComment("inferred column type");
+        cols.add(fs);
+      }
+
+      if (cols.size() == 0) {
+        // table does not exists or could not be fetched
+        return null;
+      }
+
+      table = buildTableFromColsList(tableName, cols);
+      //Setting the table properties.
+      table.getParameters().put(JDBC_DATABASE_TYPE, this.type);
+      table.getParameters().put(JDBC_DRIVER, this.driverClassName);
+      table.getParameters().put(JDBC_TABLE, tableName);
+      table.getParameters().put(JDBC_URL, this.jdbcUrl);
+      table.getParameters().put(hive_metastoreConstants.META_TABLE_STORAGE, JDBC_HIVE_STORAGE_HANDLER_ID);
+      table.getParameters().put("EXTERNAL", "TRUE");
+      Map<String, String> connectorParams = connector.getParameters();
+      for (String param: connectorParams.keySet()) {
+        if (param.startsWith(JDBC_CONFIG_PREFIX)) {
+          table.getParameters().put(param, connectorParams.get(param));
+        }
+      }
+      return table;
+    } catch (Exception e) {
+      LOG.warn("Exception retrieving remote table " + scoped_db + "." + tableName + " via data connector "
+          + connector.getName());
+      throw new MetaException("Error retrieving remote table:" + e);
+    } finally {
+      try {
+        if (rs != null) {
+          rs.close();
+        }
+      } catch (Exception ex) { /* ignore */ }
+    }
+  }
+
+  private ResultSet fetchTableViaDBMetaData(String tableName) {
+    ResultSet rs = null;
+    try {
+      rs = getConnection().getMetaData().getColumns(scoped_db, null, tableName, null);
+    } catch (SQLException sqle) {
+      LOG.warn("Could not retrieve column names from JDBC table, cause:" + sqle.getMessage());
```

Comment:
Should we not throw the exception forward instead of returning a null result set? It would be difficult to distinguish the absence of the table from a JDBC access exception, right?

Reply:
fixed.